// npmvctestmodule/src/mediator/SomeMediator.js
/**
* SomeMediator.js 
* 28-07-2015 13:24
* 
* @author 
* @email undefined
* 
*/

module.exports = function(include, puremvc) {
/**
 * @class com.domain.npmvctestmodule.mediator.SomeMediator
 * @extends puremvc.Mediator
 *
 */
puremvc.define(
	// CLASS INFO
	{
		name: "com.domain.npmvctestmodule.mediator.SomeMediator",
		parent: puremvc.Mediator
	},

	// INSTANCE MEMBERS
	{
		/** @override */
		listNotificationInterests: function() {
			return []
		},
		/** @override */
		handleNotification: function(note) {
			switch (note.getName()) {
				case "dosomething":
					//DO SOMETHING
					break;
			}
		},
		/**
		 * [init description]
		 * @return {[type]} [description]
		 */
		init: function() {
			return com.domain.npmvctestmodule.mediator.SomeMediator.NAME;
		},
		/** @override */
		onRegister: function() {
			
		},
		/** @override */
		onRemove: function() {
			
		}
	},
	// STATIC MEMBERS
	{
		/**
		 * @static
		 * @type {string}
		 */
		NAME: 'com.domain.npmvctestmodule.mediator.SomeMediator'
	});
}
